common/reflect_resource_plugin_framework.go

```diff
@@ -0,0 +1,505 @@
+
+	if !destField.IsValid() {
+		// Skip field that destination struct does not have.
+		return nil
+	}
+
+	if !destField.CanSet() {
+		panic(fmt.Errorf("destination field can not be set: %s", destField.Type().Name()))
+	}
+
+	srcFieldValue := srcField.Interface()
+
+	if srcFieldValue == nil {
+		return nil
+	}
+
+	switch srcField.Kind() {
+	case reflect.Ptr:
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		destField.Set(reflect.New(destField.Type().Elem()))
+
+		// Recursively populate the nested struct.
+		if err := GoSdkToTfSdkStruct(srcFieldValue, destField.Interface(), ctx); err != nil {
+			return err
+		}
+	case reflect.Bool:
+		boolVal := srcFieldValue.(bool)
+		// check if alwaysAdd is false or the value is zero or if the field is in the forceSendFields list
+		if alwaysAdd || !(!boolVal && !checkTheStringInForceSendFields(srcFieldName, forceSendField)) {
+			destField.Set(reflect.ValueOf(types.BoolValue(boolVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.BoolNull()))
+		}
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		// convert any kind of integer to int64
+		intVal := srcField.Convert(reflect.TypeOf(int64(0))).Interface().(int64)
+		// check if alwaysAdd is true or the value is zero or if the field is in the forceSendFields list
+		if alwaysAdd || !(intVal == 0 && !checkTheStringInForceSendFields(srcFieldName, forceSendField)) {
+			destField.Set(reflect.ValueOf(types.Int64Value(int64(intVal))))
+		} else {
+			destField.Set(reflect.ValueOf(types.Int64Null()))
+		}
+	case reflect.Float32, reflect.Float64:
+		// convert any kind of float to float64
+		float64Val := srcField.Convert(reflect.TypeOf(float64(0))).Interface().(float64)
+		// check if alwaysAdd is true or the value is zero or if the field is in the forceSendFields list
+		if alwaysAdd || !(float64Val == 0 && !checkTheStringInForceSendFields(srcFieldName, forceSendField)) {
+			destField.Set(reflect.ValueOf(types.Float64Value(float64Val)))
+		} else {
+			destField.Set(reflect.ValueOf(types.Float64Null()))
+		}
+	case reflect.String:
+		var strVal string
+		if srcField.Type().Name() != "string" {
+			// This case is for Enum Types.
+			var stringMethod reflect.Value
+			if srcField.CanAddr() {
+				stringMethod = srcField.Addr().MethodByName("String")
+			} else {
+				// Create a new addressable variable to call the String method
+				addr := reflect.New(srcField.Type()).Elem()
+				addr.Set(srcField)
+				stringMethod = addr.Addr().MethodByName("String")
+			}
+			if stringMethod.IsValid() {
+				stringResult := stringMethod.Call(nil)
+				if len(stringResult) == 1 {
+					strVal = stringResult[0].Interface().(string)
+				} else {
+					panic("num get string has more than one result")
+				}
+			} else {
+				panic("enum does not have valid .String() method")
+			}
```

Comment:
I've tried this and I don't think it works that easily because the `String()` method are on pointer types so we need to do some more `Addr` and `CanAddr` checks at the beginning, the code will actually look more complex that way